docs/src/filters/firewall.md

```diff
@@ -0,0 +1,96 @@
+# Firewall
+
+The `Firewall` filter's job is to allow or block traffic depending on if the incoming traffic's IP and port matches
+the rules set on the Firewall filter.
+
+#### Filter name
+```text
+quilkin.extensions.filters.firewall.v1alpha1.Firewall
+```
+
+### Configuration Examples
+```rust
+# let yaml = "
+version: v1alpha1
+static:
+  filters:
+    - name: quilkin.extensions.filters.firewall.v1alpha1.Firewall
+      config:
+        on_read:
+          - action: ALLOW
+            source: 192.168.51.0/24
+            ports:
+               - 10
+               - 1000-7000
+        on_write: 
+          - action: DENY
+            source: 192.168.51.0/24
+            ports:
+               - 7000
+  endpoints:
+    - address: 127.0.0.1:7001
+# ";
+# let config = quilkin::config::Config::from_reader(yaml.as_bytes()).unwrap();
+# assert_eq!(config.source.get_static_filters().unwrap().len(), 1);
+# quilkin::Builder::from(std::sync::Arc::new(config)).validate().unwrap();
+```
+
+### Configuration Options
```

Comment:
Should this just link the `Config` struct in `rustdoc` instead?

Reply:
Both seems good! Then if you know Rust, you can look there, but if you don't you have an alternate documentation:

![image](https://user-images.githubusercontent.com/298370/140432559-dce1f224-3b51-4939-92d8-b41c533eb341.png)

What do you think of that format? I must say, I quite like having both, gives you a few ways of working out what is possible depending on how you are interacting with Quilkin.

Obligatory:

![200](https://user-images.githubusercontent.com/298370/140432626-f754f77e-3f90-44c9-9642-39391e628400.gif)